### PR TITLE
[WIP] Add CSV parser

### LINKED ISSUE #131: Convert .bin files to .csv
Add a tool to export log files to csv

### PR DESCRIPTION
**Pull request by patrickelectric**
Fix #131 
Needs [#61](https://github.com/bluerobotics/ping-protocol-cpp/pull/61)

### CODE REVIEW

**Comment by patrickelectric** [comment]
Needs @brianhBR approval.

File: src/sensor/ping.cpp [hunk 1]
```diff
@@ -444,3 +447,92 @@ void Ping::writeMessage(const PingMessage &msg)
...
+        // This will avoid redundant emition
+        if(lastPercentage != percentage) {
+            lastPercentage = percentage;
+            emit csvProgress(percentage);
+        }
+
+        //Check for valid checksum and id
+        ping_msg_ping1D_empty ping_message{(uint8_t*)(packages[i].data()), (uint16_t)packages[i].length()};
+        if(!ping_message.verifyChecksum()) {
+            continue;
```
Comment (jaxxzer) [inline src/sensor/ping.cpp]: we should warn here.

**Comment by jaxxzer** [comment]
I'm wondering what's the most useful thing here. For example, if we receive distance info from two different messages, the data lands in two different files. Perhaps we should stick to only pertinent information (confidence, range, profile) and interleave this data from all messages into one output file. Meanwhile, we can forgo exporting things like firmware version, voltage, scan range etc. This consideration should not hold up this PR, however.

**Comment by patrickelectric** [comment]
> I'm wondering what's the most useful thing here. For example, if we receive distance info from two different messages, the data lands in two different files. Perhaps we should stick to only pertinent information (confidence, range, profile) and interleave this data from all messages into one output file. Meanwhile, we can forgo exporting things like firmware version, voltage, scan range etc. This consideration should not hold up this PR, however.

I would like to add each variable in a different tab in excel, and in the end only one file with a bunch of tabs. But this will need second party libraries or new classes to handle everything. This is a simple and good first version.

> Why are all the function mappings returning floats?
Is it because the functions in the map must all have the same type?

Exactly:
https://github.com/bluerobotics/ping-protocol-cpp/pull/61/files#diff-ebcbbb8c54e652539da8ea1febd4bf6eR34

**Comment by jaxxzer** [comment]
Add timestamps or open an issue for it after merge.

**Comment by patrickelectric** [comment]
Closing due to inactivity.

**Pull request closed without merging**